### PR TITLE
Python: Update chat_gpt_api.py

### PR DESCRIPTION
**Pull request by Sarfaraz021**
there are some errors in chat.add_user_message/assistant_message. You cannot just make chat as a object of ChatHistory, instead make it as 

chat_history = ChatHistory(system_message=system_message)

also define service_id and pass it into kernel.add_service: such as :

service_id = "chat-gpt"

kernel.add_service(
    sk_oai.OpenAIChatCompletion(
        service_id="chat-gpt", ai_model_id="gpt-3.5-turbo", api_key=api_key)
)

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

File: python/samples/kernel-syntax-examples/chat_gpt_api.py [hunk 1]
```diff
@@ -25,10 +29,12 @@
 
 api_key, org_id = sk.openai_settings_from_dot_env()
 kernel.add_service(
-    sk_oai.OpenAIChatCompletion(service_id="chat-gpt", ai_model_id="gpt-3.5-turbo", api_key=api_key, org_id=org_id)
+    sk_oai.OpenAIChatCompletion(
+        service_id="chat-gpt", ai_model_id="gpt-3.5-turbo", api_key=api_key)
 )
-
-settings = kernel.get_prompt_execution_settings_from_service(ChatCompletionClientBase, "chat-gpt")
```
Comment (eavanvalkenburg) [inline python/samples/kernel-syntax-examples/chat_gpt_api.py:31]: There is something for this, just needs the arguments the other way around and add _id: `settings = kernel.get_prompt_execution_settings_from_service_id("chat-gpt", ChatCompletionClientBase)`

File: python/samples/kernel-syntax-examples/chat_gpt_api.py [hunk 2]
```diff
@@ -11,6 +11,10 @@
 from semantic_kernel.functions.kernel_arguments import KernelArguments
 from semantic_kernel.prompt_template.input_variable import InputVariable
 from semantic_kernel.prompt_template.prompt_template_config import PromptTemplateConfig
+# import os
```
Comment (eavanvalkenburg) [inline python/samples/kernel-syntax-examples/chat_gpt_api.py:14]: please delete these commented rows.

**Review by eavanvalkenburg — CHANGES_REQUESTED** [review]
thanks for doing this, some minor comments!

File: python/samples/kernel-syntax-examples/chat_gpt_api.py (hunk 2 above)
Comment (Sarfaraz021) [reply python/samples/kernel-syntax-examples/chat_gpt_api.py:14]: oh sure

File: python/samples/kernel-syntax-examples/chat_gpt_api.py (hunk 1 above)
Comment (Sarfaraz021) [reply python/samples/kernel-syntax-examples/chat_gpt_api.py:31]: yes you are right, we can do this as well.

**Comment by Sarfaraz021** [comment]
> @Sarfaraz021 please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement

@microsoft-github-policy-service agree company="Xflow"

**Comment by Sarfaraz021** [comment]
..

**Review by moonbox3 — APPROVED** [review]

**Comment by moonbox3** [comment]
@Sarfaraz021 Lint is failing due to: samples/kernel-syntax-examples/chat_gpt_api.py:8:5: F401 [*] `semantic_kernel.connectors.ai.chat_completion_client_base.ChatCompletionClientBase` imported but unused

**Comment by markwallace-microsoft** [comment]
<a href="https://github.com/microsoft/semantic-kernel/blob/2ee86c1c7323b200f89ef65fbb8edf68c168feaa/README.md">Py3.8 Test Coverage</a><details><summary>Python 3.8 Test Coverage Report • </summary><table><tr><th>File</th><th>Stmts</th><th>Miss</th><th>Cover</th><th>Missing</th></tr><tbody><tr><td><b>TOTAL</b></td><td><b>5044</b></td><td><b>990</b></td><td><b>80%</b></td><td>&nbsp;</td></tr></tbody></table><i>report-only-changed-files is enabled. No files were changed during this commit :)</i></details>

## Python 3.8 Unit Test Overview
| Tests | Skipped | Failures | Errors | Time |
| ----- | ------- | -------- | -------- | ------------------ |
| 1059 | 11 :zzz: | 0 :x: | 0 :fire: | 22.161s :stopwatch: |

File: python/samples/kernel-syntax-examples/chat_gpt_api.py (hunk 1 above)
Comment (eavanvalkenburg) [reply python/samples/kernel-syntax-examples/chat_gpt_api.py:31]: could you please revert this change, I want the samples to be as simple and clear as possible!

Comment (Sarfaraz021) [reply python/samples/kernel-syntax-examples/chat_gpt_api.py:31]: sure, am gonna revert it as earlier i did

File: python/samples/kernel-syntax-examples/chat_gpt_api.py [hunk 3]
```diff
@@ -25,10 +25,11 @@
...
 api_key, org_id = sk.openai_settings_from_dot_env()
 kernel.add_service(
-    sk_oai.OpenAIChatCompletion(service_id="chat-gpt", ai_model_id="gpt-3.5-turbo", api_key=api_key, org_id=org_id)
+    sk_oai.OpenAIChatCompletion(
+        service_id="chat-gpt", ai_model_id="gpt-3.5-turbo", api_key=api_key)
 )
-
-settings = kernel.get_prompt_execution_settings_from_service(ChatCompletionClientBase, "chat-gpt")
+service_id = "chat-gpt"
+settings = kernel.get_prompt_execution_settings_from_service_id("chat-gpt", ChatCompletionClientBase)
```
Comment (eavanvalkenburg) [inline python/samples/kernel-syntax-examples/chat_gpt_api.py:32]: why not use the service_id variable you just created, and maybe move that to the line 26, and use it there as well!

**Review by eavanvalkenburg — CHANGES_REQUESTED** [review]
one small thing, really close!

**Comment by Sarfaraz021** [comment]
> @Sarfaraz021 Lint is failing due to: samples/kernel-syntax-examples/chat_gpt_api.py:8:5: F401 [*] `semantic_kernel.connectors.ai.chat_completion_client_base.ChatCompletionClientBase` imported but unused

@moonbox3 it's been used here according to the new commit suggested by @eavanvalkenburg  -> settings = kernel.get_prompt_execution_settings_from_service_id("chat-gpt", ChatCompletionClientBase)

File: python/samples/kernel-syntax-examples/chat_gpt_api.py (hunk 3 above)
Comment (Sarfaraz021) [reply python/samples/kernel-syntax-examples/chat_gpt_api.py:32]: okay sure

Comment (Sarfaraz021) [reply python/samples/kernel-syntax-examples/chat_gpt_api.py:32]: am doing some changes into code to make it bit clean

**Comment by moonbox3** [comment]
> > @Sarfaraz021 Lint is failing due to: samples/kernel-syntax-examples/chat_gpt_api.py:8:5: F401 [*] `semantic_kernel.connectors.ai.chat_completion_client_base.ChatCompletionClientBase` imported but unused
> 
> @moonbox3 it's been used here according to the new commit suggested by @eavanvalkenburg -> settings = kernel.get_prompt_execution_settings_from_service_id("chat-gpt", ChatCompletionClientBase)

I understand. If Lint says it's failing, then we need to look at why. There's no subjectivity involved. :)

**Comment by Sarfaraz021** [comment]
> > > @Sarfaraz021 Lint is failing due to: samples/kernel-syntax-examples/chat_gpt_api.py:8:5: F401 [*] `semantic_kernel.connectors.ai.chat_completion_client_base.ChatCompletionClientBase` imported but unused
> > 
> > 
> > @moonbox3 it's been used here according to the new commit suggested by @eavanvalkenburg -> settings = kernel.get_prompt_execution_settings_from_service_id("chat-gpt", ChatCompletionClientBase)
> 
> I understand. If Lint says it's failing, then we need to look at why. There's no subjectivity involved. :)

what could be the solution of this now? should I remove import st: of ChatCompletionClientBase and use previous technique

**Comment by moonbox3** [comment]
> > > > @Sarfaraz021 Lint is failing due to: samples/kernel-syntax-examples/chat_gpt_api.py:8:5: F401 [*] `semantic_kernel.connectors.ai.chat_completion_client_base.ChatCompletionClientBase` imported but unused
> > > 
> > > 
> > > @moonbox3 it's been used here according to the new commit suggested by @eavanvalkenburg -> settings = kernel.get_prompt_execution_settings_from_service_id("chat-gpt", ChatCompletionClientBase)
> > 
> > 
> > I understand. If Lint says it's failing, then we need to look at why. There's no subjectivity involved. :)
> 
> what could be the solution of this now? should I remove import st: of ChatCompletionClientBase and use previous technique

Now that you have the arguments in the correct order, it should be fine. You should try running `black .` (pip version == 24.2.0) in the semantic-kernel/python dir to make sure everything is formatted properly.

**Review by eavanvalkenburg — APPROVED** [review]
Looks good, thanks for picking up my small nits!

**Comment by eavanvalkenburg** [comment]
> > > > > @Sarfaraz021 Lint is failing due to: samples/kernel-syntax-examples/chat_gpt_api.py:8:5: F401 [*] `semantic_kernel.connectors.ai.chat_completion_client_base.ChatCompletionClientBase` imported but unused
> > > > 
> > > > 
> > > > @moonbox3 it's been used here according to the new commit suggested by @eavanvalkenburg -> settings = kernel.get_prompt_execution_settings_from_service_id("chat-gpt", ChatCompletionClientBase)
> > > 
> > > 
> > > I understand. If Lint says it's failing, then we need to look at why. There's no subjectivity involved. :)
> > 
> > 
> > what could be the solution of this now? should I remove import st: of ChatCompletionClientBase and use previous technique
> 
> Now that you have the arguments in the correct order, it should be fine. You should try running `black .` (pip version == 24.2.0) in the semantic-kernel/python dir to make sure everything is formatted properly.

or use the `Python: Run Checks` task in vscode (when you have vscode open in the python directory)!

**Comment by Sarfaraz021** [comment]
> > > > > > @Sarfaraz021 Lint is failing due to: samples/kernel-syntax-examples/chat_gpt_api.py:8:5: F401 [*] `semantic_kernel.connectors.ai.chat_completion_client_base.ChatCompletionClientBase` imported but unused
> > > > > 
> > > > > 
> > > > > @moonbox3 it's been used here according to the new commit suggested by @eavanvalkenburg -> settings = kernel.get_prompt_execution_settings_from_service_id("chat-gpt", ChatCompletionClientBase)
> > > > 
> > > > 
> > > > I understand. If Lint says it's failing, then we need to look at why. There's no subjectivity involved. :)
> > > 
> > > 
> > > what could be the solution of this now? should I remove import st: of ChatCompletionClientBase and use previous technique
> > 
> > 
> > Now that you have the arguments in the correct order, it should be fine. You should try running `black .` (pip version == 24.2.0) in the semantic-kernel/python dir to make sure everything is formatted properly.
> 
> or use the `Python: Run Checks` task in vscode (when you have vscode open in the python directory)!

sure am gonna do that and will get back asap:)

**Comment by Sarfaraz021** [comment]
> > > > > > @Sarfaraz021 Lint is failing due to: samples/kernel-syntax-examples/chat_gpt_api.py:8:5: F401 [*] `semantic_kernel.connectors.ai.chat_completion_client_base.ChatCompletionClientBase` imported but unused
> > > > > 
> > > > > 
> > > > > @moonbox3 it's been used here according to the new commit suggested by @eavanvalkenburg -> settings = kernel.get_prompt_execution_settings_from_service_id("chat-gpt", ChatCompletionClientBase)
> > > > 
> > > > 
> > > > I understand. If Lint says it's failing, then we need to look at why. There's no subjectivity involved. :)
> > > 
> > > 
> > > what could be the solution of this now? should I remove import st: of ChatCompletionClientBase and use previous technique
> > 
> > 
> > Now that you have the arguments in the correct order, it should be fine. You should try running `black .` (pip version == 24.2.0) in the semantic-kernel/python dir to make sure everything is formatted properly.
> 
> or use the `Python: Run Checks` task in vscode (when you have vscode open in the python directory)!

working fine now.

**Comment by Sarfaraz021** [comment]
Hi @moonbox3 @eavanvalkenburg , when i will be listed in the contributors?

**Review by eavanvalkenburg — APPROVED** [review]
Looks good, thanks!

**Comment by Sarfaraz021** [comment]
> @Sarfaraz021 still got a lint error for the black check...

@eavanvalkenburg let me check

**Comment by Sarfaraz021** [comment]
> @Sarfaraz021 still got a lint error for the black check...

Hi @eavanvalkenburg am done, please start testing again.

**Comment by moonbox3** [comment]
@Sarfaraz021 I am not seeing any new commit. Did you push the Lint fix upstream?

**Comment by Sarfaraz021** [comment]
> Now that you have the arguments in the correct order, it should be fine. You should try running `black .` (pip version == 24.2.0) in the semantic-kernel/python dir to make sure everything is formatted properly.

have followed these comments of yours; "Now that you have the arguments in the correct order, it should be fine. You should try running black . (pip version == 24.2.0) in the semantic-kernel/python dir to make sure everything is formatted properly." and have done with this process in my vs code : 
image

**Comment by moonbox3** [comment]
> > Now that you have the arguments in the correct order, it should be fine. You should try running `black .` (pip version == 24.2.0) in the semantic-kernel/python dir to make sure everything is formatted properly.
> > @moonbox3
> > have followed these comments of yours; "Now that you have the arguments in the correct order, it should be fine. You should try running black . (pip version == 24.2.0) in the semantic-kernel/python dir to make sure everything is formatted properly." and have done with this process in my vs code :
> > image

When you run the VSCode task or black formatting manually (via `black .`) in the `semantic-kernel/python` directory, you will see that the file `chat_gpt_api.py` will change as its formatting has updated. You then need to push the change up so the PR actions can re-run.

**Comment by Sarfaraz021** [comment]
> When you run the VSCode task or black formatting manually (via `black .`) in the `semantic-kernel/python` directory, you will see that the file `chat_gpt_api.py` will change as its formatting has updated. You then need to push the change up so the PR actions can re-run.

facing this error while pushing 
image

**Comment by Sarfaraz021** [comment]
Hi @moonbox3 @eavanvalkenburg any updates

**Comment by eavanvalkenburg** [comment]
> Hi @moonbox3 @eavanvalkenburg any updates

we can't do anything until all the checks pass, and the black check still doesn't, I would pull the latest, redo `poetry install --extras all` and then run the pre-commit checks on all files.

**Comment by moonbox3** [comment]
@Sarfaraz021 simply doing a merge from main to your PR branch isn't going to fix the lint issue. You need to make sure you pull all latest code down to your local branch on your machine, run black/ruff formatting, and then commit those changes back upstream. Only when that is done can we check if the PR is ready to be merged.

**Comment by Sarfaraz021** [comment]
> @Sarfaraz021 simply doing a merge from main to your PR branch isn't going to fix the lint issue. You need to make sure you pull all latest code down to your local branch on your machine, run black/ruff formatting, and then commit those changes back upstream. Only when that is done can we check if the PR is ready to be merged.

okay @moonbox3 let me try it accordingly

**Comment by Sarfaraz021** [comment]
image

**Comment by Sarfaraz021** [comment]
image

**Comment by Sarfaraz021** [comment]
image

**Comment by Sarfaraz021** [comment]
still 1 failing i don't know why?

**Comment by moonbox3** [comment]
image

These are the formatting changes required. I ran `black .` in the Python semantic-kernel dir, and this is the result. You then need to commit these changes. That's it.

**Comment by moonbox3** [comment]
@Sarfaraz021 I just commited the formatting fix for you.

**Comment by Sarfaraz021** [comment]
Hi there @moonbox3  @eavanvalkenburg , I tried to commit the changes but now there seem some conflicts,

**Comment by moonbox3** [comment]
We got this handled in a separate PR (#5596). We appreciate your help on initiating the PR/contribution.

**Comment by Sarfaraz021** [comment]
> We got this handled in a separate PR (#5596). We appreciate your help on initiating the PR/contribution.

thanks @moonbox3, but I checked that still am not added to the contribution list of the semantic kernel.
Please let me know when I will added to the contributors list?

**Pull request closed without merging**